bundles/org.openhab.binding.danfossairunit/src/main/java/org/openhab/binding/danfossairunit/internal/DanfossAirUnit.java

```diff
@@ -0,0 +1,278 @@
+        return communicationController.sendRobustRequest(operation, register)[0] != 0;
+    }
+
+    private void setSetting(byte[] register, boolean value) throws IOException {
+        setSetting(register, value ? (byte) 1 : (byte) 0);
+    }
+
+    private short getWord(byte[] operation, byte[] register) throws IOException {
+        byte[] resultBytes = communicationController.sendRobustRequest(operation, register);
+        return (short)((resultBytes[0] << 8)  | (resultBytes[1] & 0xFF));
+    }
+
+    private byte getByte(byte[] operation, byte[] register) throws IOException {
+        return communicationController.sendRobustRequest(operation, register)[0];
+    }
+
+    private String getString(byte[] operation, byte[] register) throws IOException {
+        // length of the string is stored in the first byte
+        byte[] result = communicationController.sendRobustRequest(operation, register);
+        return new String(result, 1, result[0], StandardCharsets.US_ASCII);
+    }
+
+    private void set(byte[] operation, byte[] register, byte value) throws IOException {
+        byte[] valueArray = { value };
+        communicationController.sendRobustRequest(operation, register, valueArray);
+    }
+
+    private void set(byte[] operation, byte[] register, short value) throws IOException {
+        communicationController.sendRobustRequest(operation, register, shortToBytes(value));
+    }
+
+    private byte[] shortToBytes(short s) {
+        return new byte[]{(byte)((s & 0xFF00) >> 8),(byte)(s & 0x00FF)};
+    }
+
+    private short getShort(byte[] operation, byte[] register) throws IOException {
+        byte[] result = communicationController.sendRobustRequest(operation, register);
+        return (short) ((result[0] << 8) + (result[1] & 0xff));
+    }
+
+    private float getTemperature(byte[] operation, byte[] register) throws IOException {
+        short shortTemp = getShort(operation, register);
+        float temp = ((float) shortTemp) / 100;
+        if (temp <= -274 || temp > 100) {
+            throw new IllegalArgumentException(String.format("Invalid temperature: %s", temp));
+        }
+        return temp;
+    }
+
+    private ZonedDateTime getTimestamp(byte[] operation, byte[] register) throws IOException {
+        byte[] result = communicationController.sendRobustRequest(operation, register);
+        return asZonedDateTime(result);
+    }
+
+    private ZonedDateTime asZonedDateTime(byte[] data) {
+        int second = data[0];
+        int minute = data[1];
+        int hour = data[2] & 0x1f;
+        int day = data[3] & 0x1f;
+        int month = data[4];
+        int year = data[5] + 2000;
+        try {
+            return ZonedDateTime.of(year, month, day, hour, minute, second, 0, ZoneId.systemDefault());
+        } catch (DateTimeException e) {
+            logger.debug("Ignoring invalid timestamp {}.{}.{} {}:{}:{}", day, month, year, hour, minute, second);
+            throw new IllegalArgumentException();
```

Comment:
I wouldn't throw an IllegalArgumentException. Similar to reasons I mentioned above. This code is used to process response messages and as such it should be a checked exception like an IOException.